### DISCUSSION TITLE
using or adding to the download queue?

### DISCUSSION THREAD

**Discussion by KaterMcDuck** [BlafKing/sd-civitai-browser-plus#200] [Q&A]
I dont know if I am missing something.  Is there a way to add more to an active queue?  I can search a page,  add what I want from that page all at once, but when I go to next page, or search for something else, I can't add anything until ALL of the current downloads are finished.i dont know if I am just going about this wrong or what.

**Comment by BlafKing** [comment]
You're right! thanks for pointing that out.

The extension currently checks to see if a download is already in the queue by looking if only the selected version name matches any entries in the queue, which is obviously a bad way of doing it since we also need to check if the selected model itself is already present.

I'll make it so that it checks if the selected model ID and version name are found instead, which should resolve this issue and only disable the button for models which are actually already in the queue.

I'll make sure to add this in the upcoming update! :)

**Reply by KaterMcDuck** [reply]
Okay cool.  I wasn't sure if I was completely missing something with it or not.    But at least I know I'm not just going blind or something now!

I don't know how easy this would be to implement, but I'd love to be able to actually see the queue, and possibly delete parts or pause them and rearrange it.   On rather slow DSL, so stuff like that makes life much easier...

But thanks!

**Reply by BlafKing** [reply]
All good!

And I'm actually currently working on a manageable download queue!
Turns out to be pretty tricky to implement, but there's progress nonetheless! 

https://github.com/BlafKing/sd-civitai-browser-plus/assets/9644716/e39b4149-36fb-4b2f-acdf-54b0ee70eb28

**Comment by BlafKing** [comment]
I've now published a new version with the new queue system included!

I didn't have time to thoroughly test everything, so there are likely still some bugs present.

(Going on holiday now so wanted to publish something since I won't be able to work on it for a little while)

**Reply by KaterMcDuck** [reply]
well dang,  that came along a lot faster than I ever could have hoped for!   I definitely appreciate the work put into it!  Thank you!
Ill keep a look out for any bugs and such I come across for you, dunno if I should put those somewhere else, or just add on to here though.

**Reply by KaterMcDuck** [reply]
One thing Ive notice with the queue,  at least for me,  when I add a bunch from one page, and then move to another page,  I can no longer "add all selected" to the queue.  doesnt matter if its 2 or 20.  have made sure none are already added.   after the first batch put into the queue, I have to manually go to each item to add to queue one by one.       

I THINK that when the queue finishes all current items,  that you can then use the add all button again, but only sometimes?  Its been weird with that part.   though it seems sometimes you have to mess around with it,  clicking the same item a couple times, or a checkbox.  Specially if you already have something checked when all other items finish downloading.

**Reply by BlafKing** [reply]
Thanks for letting me know!

I'll make sure to look into that soon :)

**Comment by KaterMcDuck** [comment]
Aye!  This one isn't something urgent.   For me at least, considering I download at around 800 kb/s, a full page can take many hours...

Which on that note,  would scheduled downloads be at all possible?       Since I have to throttle anything I do during the day when others are up and online as well, I do most downloads at night.  I know there's probably not a lot of people that need to worry about that, to make it any priority if at all.

But if you're on vacation, go do vacation shit.  Come back to this stuff when you're done with that.
